### PR TITLE
pin-2210: Improve error handling and logging

### PR DESCRIPTION
**Pull request by galales**

### CODE REVIEW

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/authorizationserver/api/impl/AuthApiServiceImpl.scala [hunk 1]
```diff
@@ -71,10 +70,10 @@ final case class AuthApiServiceImpl(
       clientUUID             <- clientId.traverse(_.toUUID)
       clientAssertionRequest <- ValidClientAssertionRequest
         .from(clientAssertion, clientAssertionType, grantType, clientUUID)
-        .recoverWith { case err: InvalidAccessTokenRequest => Failure(InvalidAssertion(err.errors.mkString(","))) }
+        .adaptError { case err: InvalidAccessTokenRequest => InvalidAssertion(err.errors.mkString(",")) }
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/authorizationserver/api/impl/AuthApiServiceImpl.scala:73]: It's better to do not use cats methods on Future, we can't safely say where this adaptError is going to be executed. To get this syntax a typeclass instance is instantiated using an implicit ec, and we can't safely say where it comes from. In general, using cats syntax on Future should be avoided.

Comment (galales) [reply src/main/scala/it/pagopa/interop/authorizationserver/api/impl/AuthApiServiceImpl.scala:73]: 👌

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/authorizationserver/api/impl/AuthApiServiceImpl.scala (hunk 1 above)
Comment (beetlecrunch) [reply src/main/scala/it/pagopa/interop/authorizationserver/api/impl/AuthApiServiceImpl.scala:73]: I approve this PR, but this comment is still open, isn't it?

**Review by beetlecrunch — APPROVED** [review]

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/authorizationserver/api/impl/AuthApiServiceImpl.scala (hunk 1 above)
Comment (galales) [reply src/main/scala/it/pagopa/interop/authorizationserver/api/impl/AuthApiServiceImpl.scala:73]: I removed all other occurrences, but this is used on `Try`. It should be safe, right?

**Pull request merged**